src/main/java/core/basesyntax/service/impl/DataHandlerServiceImpl.java

```diff
@@ -0,0 +1,30 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.dao.FruitDao;
+import core.basesyntax.service.DataHandlerService;
+import java.util.Map;
+
+public class DataHandlerServiceImpl implements DataHandlerService {
+    private static final String COLUMN_FRUIT = "fruit";
+    private static final String COLUMN_QUANTITY = "quantity";
+
+    @Override
+    public String dataHandler(FruitDao fruitDao) {
+        StringBuilder builder = new StringBuilder(COLUMN_FRUIT.concat(",")
+                .concat(COLUMN_QUANTITY).concat(System.lineSeparator()));
+        for (Map.Entry<String, Integer> pair : fruitDao.getAll().entrySet()) {
+            if (pair.getKey() == null || pair.getValue() == null) {
+                throw new RuntimeException("Can`t handle null data.");
+            }
+            if (pair.getKey().isBlank()) {
+                throw new RuntimeException("Can`t handle empty data.");
+            }
+            if (pair.getValue() < 0) {
+                throw new RuntimeException("Incorrect data less 0");
+            }
```

Comment:
Remove these IFs.  Storage must contain valid data. We should handle it during adding data to the Storage